lib/ControlledVocab/ControlledVocab.js

```diff
@@ -136,6 +153,30 @@ class ControlledVocab extends React.Component {
     return null;
   }
 
+  componentDidUpdate() {
+    // build a list of the ids of users who have updated CV items so
+    // we can look them up all at once
+    const { stripes, resources: { values: { records } } } = this.props;
+    if (stripes.hasPerm('ui-users.view')) {
+      // convert the list of values-plus-metadata to a de-duped list of
+      // metadata-userids
+      const userIds = [...new Set(records
+        .filter(r => r.metadata && r.metadata.updatedByUserId)
+        .map(r => `id=="${r.metadata.updatedByUserId}"`))
+      ];
+
+      // only query if we have users to look for
+      // when loading system data; the list may be empty!
+      // quiet, lint, I know what I'm doing. yes, I know this will kick
+      // off another render. THAT'S EXACTLY WHAT I'M TRYING TO DO.
+      // so helpful, you are.
+      if (!isEqual(userIds, this.state.userIds)) {
+        this.setState({ userIds }); // eslint-disable-line react/no-did-update-set-state
```

Comment:
I wonder do we need to rely on the local state here? Could we read the values from the `updaterIds` resource directly?

Reply:
You're right, reading directly from the resource is simpler.